### ISSUE TITLE
implement params to support xpath-like replacing and non-scalar values

### ISSUE THREAD

**Issue by goern** [projectatomic/atomicapp#192]
implement https://github.com/projectatomic/nulecule/issues/70 for Atomic App

**Comment by vpavlin** [comment]
I stared looking at this although the format and spec in projectatomic/nulecule#70  is not set in stone yet.. but I ran into a problem even before I really started..

```
obj = anymarkup.parse_file("examples/helloapache/artifacts/kubernetes/hello-apache-pod.json")
AnyMarkupError: AnyMarkupError: caught <type 'exceptions.ValueError'>: No JSON object could be decoded
```

well, there obviously is a JSON object, but it's not valid because 

```
                "ports": [
                    {
                        "containerPort": 80,
                        "hostPort": $hostport,
                        "protocol": "TCP"
                    }
```

notice the `$hostport` variable is not neither `int` nor it is quoted which is unacceptable for the parser:/

That basically means we cannot combine the two approaches - dollar-variable substitution and jsoin pointer.

Also note that if you try to quote `$hostport`, Kubernetes will fail as go parser is typed and expects `hostPort` to be `int` rather than `string`

Ideas?

**Comment by aweiteka** [comment]
Ugh. Use a substitution mode flag that the developer sets so we know the intent?

**Comment by vpavlin** [comment]
Notes from irc conversation on #nulecule 

```
(08:07:58 PM) vpavlin: We just say you can use either substitution or xpath in a single file, no mixing my friend!
(08:08:03 PM) vpavlin: in docs
(08:08:06 PM) vpavlin: or somewhere..
(08:08:56 PM) vpavlin: so the code workflow would be - try to load the object, if it fails, try to do the $ substitution and pass the result...
(08:11:17 PM) vpavlin: what it the xpath thing passes? Should the $ substitution still be done? Or do we consider these 2 mutually exclusive?
(08:11:55 PM) aweiteka: vpavlin: i vote for mutually exclusive
```

**Comment by dustymabe** [comment]
@vpavlin is this done?

**Comment by vpavlin** [comment]
I'll push my code tomorrow - it's not done yet, but it would be good if someone else can pick that up

**Comment by dustymabe** [comment]
Assigning to @charliedrage. @vpavlin, once you push your code he will pick it up and run with it. He may have some questions along the way.

**Comment by dustymabe** [comment]
Implemented in https://github.com/projectatomic/atomicapp/pull/366. Marking as done for now.

**Comment by dustymabe** [comment]
3a80dad merged to master in 90a1e94
